sherpa/plot/backends.py

```diff
@@ -119,7 +122,45 @@
 'capsize' : ['float', 'Size of the cap drawn at the end of the errorbars.']
 }
 
-class BaseBackend():
+
+class MetaBaseBackend(type):
+    '''Metaclass to operate a registry of backends
+
+    When any class that uses this metaclass is defined (i.e. the module is
+    imported) then the class is added to the global
+    `sherpa.plot.backends.PLOT_BACKENDS` dictionary using the class name or,
+    if defined the ``"name"`` attribute of that class as key.
+    If the key is already in use, a warning is issued.
+
+    This simple metaclass mimics the behaviour of entrypoints to some degree:
+    It is a registry of classes that provide a certain functionality.
+    However, classes are only added to this registry when they are imported,
+    so we need a manual import statement somewhere. `Entrypoints
+    <https://packaging.python.org/en/latest/specifications/entry-points/>`_
+    could do that for an *installed* package, even those outside of Sherpa.
+    However, the required `importlib.metadata` was added only in Python 3.10
+    to the standard library, so for now we prefer this home-grown solution that
+    can give Sherpa most of the functionality with no new dependencies.
+    '''
+    def __init__(cls, name, bases, dct):
+        super().__init__(name, bases, dct)
+
+        n = dct.get('name')
+        # Check if this is a manually set string
+        # It could also be a property.
+        if type(n) == str:
+            PLOT_BACKENDS[n] = cls
+        else:
+            n = name
+
+        if n not in PLOT_BACKENDS:
+            PLOT_BACKENDS[n] = cls
+        else:
+            warning(
+                f'{n} is already a registered name for {PLOT_BACKENDS[n]}: Not adding {cls}.')
```

Comment:
I think I'm missing something here.

If `n` is a string then this is going to set `PLOT_BACKENDS[n]` (line 152) and then warn because `n is in PLOT_BACKENDS` (as we've just set it) in line 159.

Reply:
fixing. Should be:
```
        if type(n) == str:
            n = cls
        else:
            n = name
```
but I used a more complex if before which allowed for `_alias_names` and classes to be passed in. I decided that's too complicated, so I removed it again, but not here. Both works, but only gives a warning for adding the same backend multiple times - and I have not yet written a test for that.

Reply:
Also, pylint likes to complain at me with a `type` check and suggests I use `isinstance(n, str)` instead.